src/core/syntax-tree/lexer.ts

```diff
@@ -37,8 +38,10 @@ export default function lexer(lexicon: Lexicon, query: string): ILexerResult {
             return {lexemes: result, valid: false, error: query};
         }
 
-        const value = (query.match(next.regexp) || [])[next.regexpMatch || 0];
-        result.push({lexeme: next, value});
+        const match = query.match(next.regexp) ?? [];
+        const value = match[next.regexpMatch || 0];
+        const flags = match[next.regexpFlags || -1];
+        result.push({lexeme: next, flags, value});
```

Comment:
From a lexicon perspective treat `i|s` prefix to applicable relational operators as "flags" that can be used to determine the exact processing to do during evaluation. `value` is the "base" operator + flags (e.g. `icontains`)